packages/orbit-components/src/Modal/index.js

```diff
@@ -323,360 +322,310 @@ ModalWrapperContent.defaultProps = {
   theme: defaultTheme,
 };
 
-const ModalCloseButton = ({ onClick, dataTest }) => {
-  const translate = useTranslate();
-  return (
-    <ButtonLink
-      onClick={onClick}
-      size="normal"
-      iconLeft={<Close />}
-      dataTest={dataTest}
-      type="secondary"
-      title={translate("button_close")}
-    />
-  );
-};
+const OFFSET = 40;
+
+const Modal = ({
+  size = SIZES.NORMAL,
+  children,
+  onClose,
+  fixedFooter = false,
+  isMobileFullPage = false,
+  preventOverlayClose = false,
+  hasCloseButton = true,
+  dataTest,
+}: Props) => {
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [scrolled, setScrolled] = React.useState<boolean>(false);
+  const [fullyScrolled, setFullyScrolled] = React.useState<boolean>(false);
+  const [hasModalSection, setHasModalSection] = React.useState<boolean>(false);
+  const [clickedModalBody, setClickedModalBody] = React.useState<boolean>(false);
+  const [fixedClose, setFixedClose] = React.useState<boolean>(false);
+  const [focusTriggered, setFocusTriggered] = React.useState<boolean>(false);
+  const [modalWidth, setModalWidth] = React.useState<number>(0);
+  const [footerHeight, setFooterHeight] = React.useState<number>(0);
+  const [firstFocusableEl, setFirstFocusableEl] = React.useState<HTMLElement | null>(null);
+  const [lastFocusableEl, setLastFocusableEl] = React.useState<HTMLElement | null>(null);
+
+  const modalContent: {| current: any | HTMLElement |} = React.useRef();
+  const modalBody: {| current: any | HTMLElement |} = React.useRef();
+  const modalID: string = randomID("modalID");
+
+  const prevChildren = usePrevious(children);
+
+  /* eslint-disable no-use-before-define */
+  React.useEffect(() => {
+    const timer: TimeoutID = setTimeout(() => {
```

Comment:
we have a hook `useStateWithTimeout` :) might be useful in this case, but it looks good already, so it's up to you